clients/src/main/java/org/apache/kafka/common/cache/Cache.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package org.apache.kafka.common.cache;
+
+/**
+ * Interface for caches, semi-peristent maps which store key-value mappings until either an eviction criteria is met
+ * or the entries are manually invalidated. Caches are not required to be thread-safe, but some implementations may be.
+ */
+public interface Cache<K, V> {
```

Comment:
I think this falls under the same category as the utils discussion. We can relocate it somewhere internally. I think we may want to extend this at some point with an implementation that also just uses reference equality -- this could actually be useful here where we'd actually prefer a really fast hash + equals check since the common pattern for connectors is to reuse the same `Schema` object.

I know Confluent's code would be able to avoid some duplication if these were publicly maintained -- our Avro converter is going to need to do similar caching. But that alone isn't necessarily a good reason to publicly support it.

I'm open to suggestions here, I don't feel too strongly one way or the other.